graphics/usd/Portfile

```
@@ -0,0 +1,218 @@
+                    applications. It organizes data into hierarchical \
+                    namespaces of primitives and their properties. \
+                    This provides a set of schemas upon which common \
+                    3-D computer graphics concepts such as geometry, \
+                    shading, models, and assets can be organized and \
+                    manipulated.
+
+checksums           rmd160  6ecba2960ecb802dd8db62344582e2d01234f3d7 \
+                    sha256  6cbd3f7be424fcb8dcc35d07f0d2cc8bc0601b2c46d06553290e6df7c5b5ec77 \
+                    size    37563045
+
+set py_ver          2.7
+set py_ver_nodot    [string map {. {}} ${py_ver}]
+
+depends_lib-append  port:boost \
+                    port:tbb
+
+compiler.cxx_standard 2011
+compiler.blacklist-append {clang < 802}
+
+# This patch adds modifications that allow Blender to use USD. The patch
+# file 'patch-add-blender-mods.diff' was extracted from the file in
+# Blender's source code named
+# 'build_files/build_environment/patches/usd.diff'. (No further
+# modifications to the patch itself were needed, since Blender 2.83 was
+# written to be compatible with USD version 19.11.) These modifications
+# essentially add a hook for Blender to use, and shouldn't otherwise
+# affect the functioning of USD.
+patchfiles-append   patch-add-blender-mods.diff
+
+# By default, USD places plugins in a ${prefix}/plugin directory. This
+# patch changes it so that the plugins get placed in a location that
+# conforms with MacPorts' mtree layout: ${prefix}/share/usd/plugins
+patchfiles-append   patch-plugins-output-paths.diff
+
+post-patch {
+    # USD puts its CMake package config files in the wrong place
+    reinplace {s|cmake/||} ${worksrcpath}/pxr/pxrConfig.cmake.in
+    reinplace -E "s|\(set.PXR_INCLUDE_DIRS \).*|\\1\"${prefix}/include\")|" \
+        ${worksrcpath}/pxr/pxrConfig.cmake.in
+    foreach re [list \
+        {s|(PROJECT_BINARY_DIR.*)(pxrConfig)|\1lib/cmake/usd/\2|} \
+        {/DESTINATION.*CMAKE_INSTALL_PREFIX/s|(DESTINATION ")(.*)"|\1\2/lib/cmake/usd"|} \
+        {/DESTINATION.*cmake/s|(DESTINATION ")cmake|\1lib/cmake/usd|} \
+    ] {
+        reinplace -E $re ${worksrcpath}/pxr/CMakeLists.txt
+    }
+}
+
+# Turn off all of USD's default build options. We will allow users to
+# directly control these options using port variants.
+configure.args      -DPXR_BUILD_TESTS=OFF \
+                    -DPXR_BUILD_IMAGING=OFF \
+                    -DPXR_BUILD_USD_IMAGING=OFF \
+                    -DPXR_BUILD_USDVIEW=OFF \
+                    -DPXR_ENABLE_GL_SUPPORT=OFF \
+                    -DPXR_ENABLE_PYTHON_SUPPORT=OFF \
+                    -DPXR_ENABLE_HDF5_SUPPORT=OFF \
+                    -DPXR_ENABLE_PTEX_SUPPORT=OFF
+
+############################# Variants ##############################
+
+variant docs description {Build documentation} {
+    depends_build-append    port:doxygen \
+                            port:graphviz
```

Specify the dependency as `path:bin/dot:graphviz` so that graphviz-devel could satisfy it.
```suggestion
                            path:bin/dot:graphviz
```

---

If I change it to `path:bin/dot:graphviz` and do the following:

```
port -f deactivate graphviz
port install graphviz-devel
```

and then run `port clean usd && port -vst install usd`, I get the following error message:

```
---> Computing dependencies for usd.
Error: Unable to execute port: Registry error: graphviz not registered as installed & active.
```

If I then do a

```
port -f deactivate graphviz-devel
port activate graphviz
```

then `port clean usd && port -vst install usd` proceeds to completion just fine. I don't mind making the change... but I just thought you should know that it doesn't seem like this change allows `graphviz-devel` to be used; at least not on my machine.